packages/control-hub/src/internal/ControlHub.ts

```diff
@@ -0,0 +1,421 @@
+import {
+    ExpansionHub,
+    ParentRevHub,
+    RevHub,
+    RevHubType,
+} from "@rev-robotics/expansion-hub";
+import {
+    BulkInputData,
+    DebugGroup,
+    DIODirection,
+    I2CReadStatus,
+    I2CSpeedCode,
+    I2CWriteStatus,
+    LEDPattern,
+    ModuleInterface,
+    ModuleStatus,
+    PIDCoefficients,
+    RGB,
+    VerbosityLevel,
+    Version,
+} from "@rev-robotics/rhsplib";
+import axios from "axios";
+import semver from "semver";
+import WebSocket from "isomorphic-ws";
+import adb from "@u4/adbkit";
+import { ControlHub, ParentControlHub } from "../ControlHub.js";
+
+export class ControlHubInternal implements ControlHub {
+    readonly isOpen: boolean = true;
+    moduleAddress: number = 0;
+    responseTimeoutMs: number = 0;
+    type: RevHubType = RevHubType.ControlHub;
+    webSocketConnection!: WebSocket;
+    private readonly serialNumber?: string;
+
+    keyGenerator = 0;
+    currentActiveCommands = new Map<
+        any,
+        (response: any | undefined, error: any | undefined) => void
+    >();
+
+    constructor(serialNumber?: string) {
+        this.serialNumber = serialNumber;
+    }
+
+    isParent(): this is ParentControlHub {
+        return this.serialNumber !== undefined;
+    }
+
+    async open(ip: string = "192.168.43.1", port: string = "8081"): Promise<void> {
```

Comment:
Given that this is only called internally, I'd rather force the caller to always specify the port and IP address.

Reply:
The CH's IP address and ports for Wi-Fi should also be defined in constants.